SPECS-EXTENDED/hdf/hdf.spec

```diff
@@ -0,0 +1,561 @@
+cd build-static
+# Java requires shared libraries, fortran requires static
+
+# Temporary workaround for compiling on GCC-10
+export FCFLAGS="%{build_fflags} -fallow-argument-mismatch"
+export FFLAGS="%{build_fflags} -fallow-argument-mismatch"
+%configure --disable-production --disable-java --disable-netcdf \
+ --enable-shared=no --enable-static=yes --enable-fortran %{!?el6:--with-szlib} \
+ --includedir=%{_includedir}/%{name}
+%make_build
+cd -
+
+# correct the timestamps based on files used to generate the header files
+touch -c -r hdf/src/hdf.inc hdf/src/hdf.f90
+touch -c -r hdf/src/dffunc.inc hdf/src/dffunc.f90
+touch -c -r mfhdf/fortran/mffunc.inc mfhdf/fortran/mffunc.f90
+# netcdf fortran include need same treatement, but they are not shipped
+
+
+%install
+%make_install -C build-static
+%make_install -C build-shared
+chrpath --delete --keepgoing %{buildroot}%{_bindir}/* %{buildroot}%{_libdir}/%{name}/*.so.* %{buildroot}%{_libdir}/*.so.* || :
+
+install -pm 644 MANIFEST README.txt release_notes/*.txt %{buildroot}%{_pkgdocdir}/
+
+find %{buildroot} -type f -name "*.la" -delete -print
+find %{buildroot} -type f -name "*.la" -delete -print
+
+#Don't conflict with netcdf
+for file in ncdump ncgen; do
+  mv %{buildroot}%{_bindir}/$file %{buildroot}%{_bindir}/h$file
+  # man pages are the same than netcdf ones
+  rm %{buildroot}%{_mandir}/man1/${file}.1
+done
+
+# this is done to have the same timestamp on multiarch setups
+touch -c -r README.txt %{buildroot}%{_includedir}/hdf/h4config.h
+
+# Remove an autoconf conditional from the API that is unused and cause
+# the API to be different on x86 and x86_64
+pushd %{buildroot}%{_includedir}/hdf
+grep -v 'H4_SIZEOF_INTP' h4config.h > h4config.h.tmp
+touch -c -r h4config.h h4config.h.tmp
+mv h4config.h.tmp h4config.h
+popd
+
+#Fixup headers and scripts for multiarch
+%if "%{_lib}" == "lib64"
+#sed -i -e s/H5pubconf.h/H5pubconf-64.h/ %{buildroot}%{_includedir}/H5public.h
+#mv %{buildroot}%{_includedir}/H5pubconf.h \
+   #%{buildroot}%{_includedir}/H5pubconf-64.h
+for x in h4cc h4fc
+do
+  mv %{buildroot}%{_bindir}/${x} \
+     %{buildroot}%{_bindir}/${x}-64
+  install -m 0755 %{SOURCE1} %{buildroot}%{_bindir}/${x}
+done
+%else
+#sed -i -e s/H5pubconf.h/H5pubconf-32.h/ %{buildroot}%{_includedir}/H5public.h
+#mv %{buildroot}%{_includedir}/H5pubconf.h \
+   #%{buildroot}%{_includedir}/H5pubconf-32.h
+for x in h4cc h4fc
+do
+  mv %{buildroot}%{_bindir}/${x} \
+     %{buildroot}%{_bindir}/${x}-32
+  install -m 0755 %{SOURCE1} %{buildroot}%{_bindir}/${x}
+done
+%endif
+
+
+%check
+make -j1 -C build-shared check
+make -j1 -C build-static check
+
+
+%files
+%license COPYING
+%{_pkgdocdir}/
+%exclude %{_pkgdocdir}/examples
+%{_bindir}/*
+%exclude %{_bindir}/h4?c*
+%{_libdir}/*.so.0*
+%{_mandir}/man1/*.gz
+
+%files devel
+%{_bindir}/h4?c*
+%{_includedir}/%{name}/
+%{_libdir}/*.so
+%{_libdir}/*.settings
+%{_pkgdocdir}/examples/
+
+%files libs
+%{_libdir}/*.so.0*
+
+%files static
+%{_libdir}/*.a
```

Comment:
We might want to take a note of this.
If this is absolutely needed; otherwise we generally do not prefer shipping static libraries.

Reply:
Ack.